libyul/AsmPrinter.h

```diff
@@ -46,12 +46,20 @@ struct Dialect;
 class AsmPrinter
 {
 public:
+	enum class Mode
+	{
+		OmitDefaultType,
```

Comment:
Maybe its better if this would be called `OmitTypeInfo`?

Reply:
I chose the name because there is a special case of bool literals which still get their type printed (provided the default type of the dialect isn't empty). So it's not omitting _all_ type info :D

Reply:
Ah ok, then that makes sense. When I was looking at the tests yesterday, I thought that all types where omitted. I missed yesterday that there was still that special `bool` case.